citeproc/model.py

```diff
@@ -373,13 +373,14 @@ def case(self, text, language=None):
                 text = ' '.join(output)
             elif text_case == 'title':
                 output = []
+                # : gurantees that the first word is capitalized
                 prev = ':'
                 for word in text.words():
-                    if not text.isupper() and not word.isupper():
+                    if (str(word) not in self._stop_words or
+                        prev in (':', '.')):
+                        word = word.capitalize_first()
+                    else:
                         word = word.soft_lower()
```

Comment:
The v1.0.1 [spec](https://docs.citationstyles.org/en/v1.0.1/specification.html#title-case-conversion) says:

> For lower or mixed case strings, the first character of each lowercase word is capitalized. The case of words in mixed or uppercase stays the same.

I suppose we should ensure the `word` is in lowercase and drop the `else` branch.

```python
                    if word.islower() and (str(word) not in self._stop_words or
                        prev in (':', '.')):
                        word = word.capitalize_first()
```